src/server/game/Entities/Creature/Creature.h

```diff
@@ -101,6 +101,14 @@ class TC_GAME_API Creature : public Unit, public GridObject<Creature>, public Ma
         bool HasReactState(ReactStates state) const { return (m_reactState == state); }
         void InitializeReactState();
 
+        // These shadow the definitions on Unit and override the default value for keepCombat
```

Comment:
they're currently not virtual, though i should probably change that (they're not called often enough that overhead matters)